src/utils/read_gcm_driven_scm_data.jl

```diff
@@ -0,0 +1,48 @@
+# Utilities for reading NetCDF data for GCM-driven single-column simulations.
+import NCDatasets as NC
+import StatsBase: mean
+
+read_gcm_driven_initial_profile_mean(FT, ds, group, varname; imin = 100) = vec(
+    mean(
+        _gcm_driven_variable_zt(FT, ds, group, varname)[:, imin:end],
+        dims = 2,
+    ),
+)
+
+# TODO: Cast to CuVector for GPU compatibility
+read_gcm_driven_initial_profile(FT, ds, group, varname) =
+    _gcm_driven_variable_zt(FT, ds, group, varname)[:, 1]  # 1 is initial time index
+
+read_gcm_driven_reference_profile(FT, ds, group, varname) =
+    _gcm_driven_variable_z(FT, ds, group, varname)[:]
+
+_gcm_driven_nz(ds) = ds.group["reference"].dim["z"]
+
+_gcm_driven_variable_zt(FT, ds, group, varname) =
+    _gcm_driven_variable(FT, ds, group, varname, ("z", "t"))
+
+_gcm_driven_variable_z(FT, ds, group, varname) =
+    _gcm_driven_variable(FT, ds, group, varname, ("z",))
+
+
+function _gcm_driven_variable(
```

Comment:
I think these 8 helper functions should be consolidated into just a few.  The get initial profile operation, for example, could be 1 line. The use case for adding ie. `_gcm_driven_variable` functions isn't clear to me, given we know the group name and they're specified elsewhere in the code. The group could maybe be a kwarg elsewhere with a default to "profiles".